src/main/scala/org/labrad/manager/Listener.scala

```diff
@@ -5,49 +5,126 @@ import io.netty.channel._
 import io.netty.channel.nio.NioEventLoopGroup
 import io.netty.channel.socket.SocketChannel
 import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.handler.ssl.{SniHandler, SslContext}
+import io.netty.util.DomainNameMapping
 import org.labrad.PacketCodec
 import org.labrad.util._
 import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
 
 /**
- * listens for incoming labrad network connections
+ * Manager policies for securing incoming client/server connections with TLS.
  */
-class Listener(auth: AuthService, hub: Hub, tracker: StatsTracker, messager: Messager, port: Int)(implicit ec: ExecutionContext)
-extends Logging {
+sealed trait TlsPolicy
+
+object TlsPolicy {
+  /**
+   * Initial connection uses TLS.
+   */
+  case object ON extends TlsPolicy
+
+  /**
+   * No TLS; connection is unencrypted and attempts to upgrade with STARTTLS
+   * will be rejected. This is provided for testing and for compatibility with
+   * the old manager.
+   */
+  case object OFF extends TlsPolicy
+
+  /**
+   * Default STARTTLS behavior. Connection starts unencrypted and we use
+   * STARTTLS to upgrade later. STARTTLS is required for remote connections,
+   * but optional for connections from localhost.
+   */
+  case object STARTTLS extends TlsPolicy
+
+  /**
+   * STARTTLS is optional for all connections, including from remote hosts.
+   */
+  case object STARTTLS_OPT extends TlsPolicy
+
+  /**
+   * STARTTLS is required for all connection, including from localhost.
+   */
+  case object STARTTLS_FORCE extends TlsPolicy
 
+  def fromString(s: String): TlsPolicy = {
+    s.toLowerCase match {
+      case "on" => ON
+      case "off" => OFF
```

Comment:
This is really only here for tests, so that things can be configured to behave like old managers that do not support TLS and verify that client connections fail. If you look at the manager configuration, there's actually no way to configure the manager to run in this mode.